napari/util/tests/test_keybindings.py

```diff
@@ -95,61 +104,183 @@ def foo():
     assert kb == dict(A=foo)
 
 
-def test_InheritedKeymap():
-    d = {}
-    k = InheritedKeymap(lambda: d, {'A': 0})
-    assert k['A'] == 0
+def test_keymap_provider():
```

Comment:
@kne42 could you split this into several tests that test different aspects of the keymap provider? The class definitions could be outside the test, or in a pytest fixture.